src/WebJobs.Extensions.OpenAI/Embeddings/EmbeddingsHelper.cs

```diff
@@ -0,0 +1,132 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Diagnostics;
+using Azure.AI.OpenAI;
+
+namespace Microsoft.Azure.WebJobs.Extensions.OpenAI.Embeddings;
+static class EmbeddingsHelper
+{
+    static readonly char[] sentenceEndingsDefault = new[] { '.', '!', '?' };
+    static readonly char[] wordBreaksDefault = new[] { ',', ';', ':', ' ', '(', ')', '[', ']', '{', '}', '\t', '\n' };
+    static readonly string UserAgent = $"{typeof(OpenAIExtension).Namespace}/{FileVersionInfo.GetVersionInfo(typeof(OpenAIExtension).Assembly.Location).FileVersion}";
+    static readonly HttpClient httpClient = new();
+
+    public static async Task<EmbeddingsOptions> BuildRequest(int maxOverlap, int maxChunkLength, string model, InputType inputType, string input)
+    {
+        using TextReader reader = await GetTextReader(inputType, input);
+        if (maxOverlap >= maxChunkLength)
+        {
+            throw new ArgumentOutOfRangeException($"MaxOverlap ({maxOverlap}) must be less than MaxChunkLength ({maxChunkLength}).");
+        }
+
+        List<string> chunks = GetTextChunks(reader, 0, maxChunkLength, maxOverlap).ToList();
+        return new EmbeddingsOptions(model, chunks);
+    }
+
+    static async Task<TextReader> GetTextReader(InputType inputType, string input)
+    {
+        if (inputType == InputType.RawText)
+        {
+            return new StringReader(input);
+        }
+        else if (inputType == InputType.FilePath)
+        {
+            return new StreamReader(input);
+        }
+        else if (inputType == InputType.Url)
+        {
+            httpClient.DefaultRequestHeaders.UserAgent.ParseAdd(UserAgent);
```

Comment:
Rather than doing this on each request, you can do it once in a static constructor.